src/components/accessories/patientTherapy/patientTherapyTable/PatientTherapyTable.tsx

```diff
@@ -13,24 +11,25 @@ const PatientTherapyTable: FunctionComponent<IOwnProps> = ({
   shouldUpdateTable,
 }) => {
   const [, setUpdate] = useState(false);
-  const { t } = useTranslation();
-  const therapy = useSelector<IState, ITherapiesState>(
-    (state) => state.therapies
+
+  let data = useSelector<IState, any[]>((state) =>
+    state.therapies.therapiesByPatientId.data
+      ? state.therapies.therapiesByPatientId.data
+      : []
   );
-  if (therapy.createTherapy.data) {
-    const newRow = {
-      startDate: therapy.createTherapy.data!.startDate!,
-      endDate: therapy.createTherapy.data!.endDate!,
-      medicalId: therapy.createTherapy.data!.medicalId! + "",
-      qty: therapy.createTherapy.data!.qty!,
-      freqInDay: therapy.createTherapy.data!.freqInDay!,
-      freqInPeriod: therapy.createTherapy.data!.freqInPeriod!,
-      unitID: therapy.createTherapy.data!.unitID!,
-      note: therapy.createTherapy.data!.note!,
-    };
-    const found = data.some((el) => el.medicalId === newRow.medicalId);
-    if (!found) data.push(newRow);
-  }
+
+  const updateTableData = (state: IState) => {
+    let newTherapyObj = state.therapies.createTherapy.data;
+    if (newTherapyObj) {
+      const found = data.find(
+        (item) => item.medicalId == newTherapyObj?.medicalId
+      );
+      if (!found) {
+        data = [...data, newTherapyObj];
+      }
+    }
+  };
+  useSelector<IState>((state) => updateTableData(state));
```

Comment:
I would put the data variable inside a useState and I'd delete the setUpdate method.

So, when the shouldUpdateTable will be updated, inside the useEffect you'll perform the Redux call updating the data with an setData.

This should be more readable, what do you think?

Reply:
It is right.

Reply:
It is not possible to use useSelector inside useEffect

Reply:
Yes, you're right. But, if I'm not wrong, you can access to the state outside the useEffect, maybe setting a variable state. Am I right or I'm just overthinking?